src/components/molecules/AdminCheckbox/AdminCheckbox.tsx

```diff
@@ -63,15 +65,19 @@ export const AdminCheckbox: React.FC<AdminCheckboxProps> = ({
     </>
   );
 
+  const isToggleFlip = variant === "flip-switch";
+  const isInputWithLabel = label || isToggleFlip;
+
   return (
     <p className={parentClasses}>
-      {label ? (
+      {isInputWithLabel ? (
         <label className="AdminCheckbox__label">
-          {labelBefore && (
-            <span className="AdminCheckbox__label--before">{labelBefore}</span>
+          {isToggleFlip && (
+            <span className="AdminCheckbox__label--before">{textOff}</span>
           )}
           {input}
           {label && label}
+          {isToggleFlip && textOn}
```

Comment:
Not sure if  messing  with the label is the right way to go. Probably leave it on its own so `textOn` and `textOff` should be visible regardless if there is a label or not. 

I'm thinking of a scenario that might have a label on top from the toggle instead from a section, but I will have to do a little local check to know for sure.

Reply:
I think there is a way to untangle this, have something done, just need to check if it works: https://github.com/sparkletown/sparkle/commit/aef7642add90dbcc95f735091b3e13bb5c68afc8#diff-1bb7cbf10026a360bed006fc8713a20180ac64d4844f4946e0444415dacf9d1b